test/lib/jdk/test/lib/process/Proc.java

```diff
@@ -282,18 +321,26 @@ public Proc start() throws IOException {
             }
         }
 
-        Collections.addAll(cmd, splitProperty("test.vm.opts"));
-        Collections.addAll(cmd, splitProperty("test.java.opts"));
-
-        if (cp == null) {
-            cmd.add("-cp");
-            cmd.add(System.getProperty("test.class.path") + File.pathSeparator +
-                    System.getProperty("test.src.path"));
-        } else if (!cp.isEmpty()) {
+        var lcp = fullcp();
+        if (lcp != null) {
             cmd.add("-cp");
-            cmd.add(cp.stream().collect(Collectors.joining(File.pathSeparator)));
+            cmd.add(lcp);
         }
 
+        if (compile) {
+            var comp = CompilerUtils.compile(
```

Comment:
I don't find this use of `var` helpful - `var` is useful for saying "some type I don't really care about and can't be bothered to spell out". But here we just have a boolean so it is clearer to just say `boolean compiled` (or `success` or `ok` or some such name that reflects what the return value actually means). Thanks.